src/icons/fileIcons.ts

```diff
@@ -1004,185 +975,44 @@ export const fileIcons: FileIcons = {
-        'webpack.config.staging.babel.js',
-        'webpack.config.staging.babel.ts',
-        'webpack.config.coffee',
-        'webpack.config.test.js',
-        'webpack.config.test.cjs',
-        'webpack.config.test.mjs',
-        'webpack.config.test.ts',
-        'webpack.config.test.cts',
-        'webpack.config.test.mts',
-        'webpack.config.vendor.js',
-        'webpack.config.vendor.cjs',
-        'webpack.config.vendor.mjs',
-        'webpack.config.vendor.ts',
-        'webpack.config.vendor.cts',
-        'webpack.config.vendor.mts',
-        'webpack.config.vendor.production.js',
-        'webpack.config.vendor.production.cjs',
-        'webpack.config.vendor.production.mjs',
-        'webpack.config.vendor.production.ts',
-        'webpack.config.vendor.production.cts',
-        'webpack.config.vendor.production.mts',
-        'webpack.test.js',
-        'webpack.test.cjs',
-        'webpack.test.mjs',
-        'webpack.test.ts',
-        'webpack.test.cts',
-        'webpack.test.mts',
-        'webpack.dist.js',
-        'webpack.dist.cjs',
-        'webpack.dist.mjs',
-        'webpack.dist.ts',
-        'webpack.dist.cts',
-        'webpack.dist.mts',
-        'webpackfile.js',
-        'webpackfile.cjs',
-        'webpackfile.mjs',
-        'webpackfile.ts',
-        'webpackfile.cts',
-        'webpackfile.mts',
-      ],
+      fileNames: ['webpack.config.coffee'],
+      patterns: {
+        'webpack.base': Ecmascript,
+        'webpack.client': Ecmascript,
+        'webpack.common': Ecmascript,
+        'webpack.config.babel': Ecmascript,
+        'webpack.config.base.babel': Ecmascript,
+        'webpack.config.base': Ecmascript,
+        'webpack.config.client': Ecmascript,
+        'webpack.config.common.babel': Ecmascript,
+        'webpack.config.common': Ecmascript,
+        'webpack.config.dev.babel': Ecmascript,
+        'webpack.config.dev': Ecmascript,
+        'webpack.config.main': Ecmascript,
+        'webpack.config.prod.babel': Ecmascript,
+        'webpack.config.prod': Ecmascript,
+        'webpack.config.production.babel': Ecmascript,
+        'webpack.config.production': Ecmascript,
+        'webpack.config.renderer': Ecmascript,
+        'webpack.config.server': Ecmascript,
+        'webpack.config.staging.babel': Ecmascript,
+        'webpack.config.staging': Ecmascript,
+        'webpack.config.test': Ecmascript,
+        'webpack.config.vendor.production': Ecmascript,
+        'webpack.config.vendor': Ecmascript,
+        'webpack.config': Ecmascript,
+        'webpack.dev': Ecmascript,
+        'webpack.development': Ecmascript,
+        'webpack.dist': Ecmascript,
+        'webpack.mix': Ecmascript,
+        'webpack.prod.config': Ecmascript,
+        'webpack.prod': Ecmascript,
+        'webpack.production': Ecmascript,
+        'webpack.server': Ecmascript,
+        'webpack.test': Ecmascript,
+        webpack: Ecmascript,
+        webpackfile: Ecmascript,
+      },
```

Comment:
How `Ecmascript` patterns are valuable, we don't need to check that 6 extensions exist for every name.

Reply:
That looks very valuable to me. I think sometimes the whole set of Cosmiconfig is too much and EcmaScript file extensions are enough. I like this approach!

Reply:
At first, I listed a lot of common patterns in the industry, but there were too many of them and it felt overwhelming, so I decided to focus on the patterns that currently exist in `fileIcons.ts`. So 4 patterns should be enough.

So what's next?

Reply:
Thank you, it looks good to me. Nothing to do from your side. It's a very helpful improvement to this extension!

Reply:
We can still add more patterns in the future if needed, so no worries about this.